sliceanddice/host/src/main/java/com/novoda/sliceanddice/SlicePermissions.kt

```diff
@@ -0,0 +1,76 @@
+package com.novoda.sliceanddice
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.app.PendingIntent
+import android.content.IntentSender
+import android.net.Uri
+import android.util.Log
+import android.widget.Toast
+import androidx.slice.Slice
+import androidx.slice.SliceManager
+
+internal fun SliceManager.missingPermission(sliceUri: Uri, appName: String) = try {
```

Comment:
The fact that we even need to do all this is because the permission slice in 28.0.0-alpha1 and P DP1 is broken. It should be clickable, but it isn't. This whole file is a pile of hacks that I hope we can get rid of in future releases (when the `LiveData` will start working, we won't need to check for permission requests etc).